### ISSUE TITLE
Unable to send data through DataConnection from iOS (Safari, Chrome), receiving works

### ISSUE THREAD

**Issue by maxpavlov** [peers/peerjs#572]
I have a tested file transfer application that follows a correct practice of establishing peer connection, the data connection, wiring up event handlers and starts the data exchange between two peers. All works like a charm on different setups in many network environments. 

However, my code loaded in iOS Safari and iOS Chrome is able to receive data via open datachannel, while it is unable to send data successfully. The connection.send executes without an error, but no actual data is received on the other end. Receiving data works, sending does not result in a sent data being received on another peer's end. No errors are observed and the same code works well on desktop versions of these browsers.

Tested with reliable true and false, no difference. Please assist in where to look at least.

**Comment by maxpavlov** [comment]
I've actually been able to fix the iOS thing by fixing the peerjs (in local repo that I am using and constantly updating). 

Thing is - in dataconnection.ts we assume that if util says binary blob is supported, we can send blobs via WebRTC. That is not true. Any attempt to send blob via RTCDataChannel from iOS will fail with unsupported action error.

I've added isiOS property to DataConnection ( decided not to change the logic of the util as it follows a quirky approach of 'if data is true, binary blob is also true'. When i know that my code is in iOS - I am sending only ArrayBuffer. It works like a charm now.

**Comment by themindstorm** [comment]
Did you previously get an error like this?
```
Unhandled Promise Rejection: TypeError: Argument 1 ('options') to RTCPeerConnection.createOffer must be a dictionary
```
PeerJS isn't working in Safari for me (MacOS and iOS) #574

**Comment by afrokick** [comment]
@maxpavlov I think we can fix it in PeerJS.

**Comment by afrokick** [comment]
@maxpavlov could you please check the latest version from master? I think it should work on iOS because fixed in #577
